python/GafferSceneUI/SetFilterUI.py

```diff
@@ -94,3 +100,162 @@
 	}
 
 )
+
+##########################################################################
+# NodeGadget drop handler
+##########################################################################
+
+GafferUI.Pointer.registerPointer( "sets", GafferUI.Pointer( "pointerSets.png", imath.V2i( 53, 14 ) ) )
+GafferUI.Pointer.registerPointer( "addSets", GafferUI.Pointer( "pointerAddSets.png", imath.V2i( 53, 14 ) ) )
+GafferUI.Pointer.registerPointer( "removeSets", GafferUI.Pointer( "pointerRemoveSets.png", imath.V2i( 53, 14 ) ) )
+GafferUI.Pointer.registerPointer( "replaceSets", GafferUI.Pointer( "pointerReplaceSets.png", imath.V2i( 53, 14 ) ) )
+
+__DropMode = enum.Enum( "__DropMode", [ "None_", "Add", "Remove", "Replace" ] )
+
+__originalDragPointer = None
+
+def __setsPlug( node ) :
+
+	for plug in node.children( Gaffer.Plug ) :
+		if Gaffer.Metadata.value( plug, "ui:scene:acceptsSetNames" ) or Gaffer.Metadata.value( plug, "ui:scene:acceptsSetExpression" ) :
+			return plug
+
+	return None
+
+def __dropMode( nodeGadget, event ) :
+
+	if __setsPlug( nodeGadget.node() ) is None :
+		filter = None
+		if nodeGadget.node()["filter"].getInput() is not None :
+			filter = nodeGadget.node()["filter"].source().node()
+		if filter is None :
+			return __DropMode.Replace
+		elif not isinstance( filter, GafferScene.SetFilter ) :
+			return __DropMode.None_
+
+	if event.modifiers & event.Modifiers.Shift :
+		return __DropMode.Add
+	elif event.modifiers & event.Modifiers.Control :
+		return __DropMode.Remove
+	else :
+		return __DropMode.Replace
+
+def __expressionSets( setExpression ) :
+
+	setTokens = setExpression.split( " " )
```

Comment:
I think we need to include `(` and `)` in the tokens for splitting, so we can detect things like `A in(C)`. Or do `"(" in setExpression` first?

Reply:
This is fixed by https://github.com/GafferHQ/gaffer/pull/5786/commits/f1ecf9410587b4a97ef766ec503d9de265ad5e88.